### ISSUE TITLE
xUnit crash with -1073740791

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#25119]
EditorServices2 crashed with `-1073740791` in a test run. My searching indicates that this code is actually C0000409, which is a stack buffer overflow code. Output:

```
00:11:19 Errors Roslyn.Services.Editor2.UnitTests.dll: 
00:11:19 
00:11:19 Command: C:\Users\dotnet-bot\.nuget\packages\xunit.runner.console\2.3.1\tools\net452\xunit.console.x86.exe "D:\j\workspace\windows_relea---fcb95091\Binaries\Release\UnitTests\EditorServicesTest2\Roslyn.Services.Editor2.UnitTests.dll"  -xml "D:\j\workspace\windows_relea---fcb95091\Binaries\Release\UnitTests\EditorServicesTest2\xUnitResults\Roslyn.Services.Editor2.UnitTests.dll.xml" -noshadow -verbose
00:11:19 xUnit output log: D:\j\workspace\windows_relea---fcb95091\Binaries\Release\UnitTests\EditorServicesTest2\xUnitResults\Roslyn.Services.Editor2.UnitTests.dll.out.log
00:11:19 
xunit produced no error output but had exit code -1073740791
```
Log File: [Roslyn.Services.Editor2.UnitTests.dll.out.log.txt](https://github.com/dotnet/roslyn/files/1768403/Roslyn.Services.Editor2.UnitTests.dll.out.log.txt)

**Comment by 333fred** [comment]
Link to the job: https://ci.dot.net/job/dotnet_roslyn/view/Official%20Builds/job/dev15.7.x-vs-deps/job/windows_release_unit32/44/

**Comment by Shyam-Gupta** [comment]
Its coming up intermittently in the PR builds. Happened today as well: https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_release_vs-integration_prtest/10467/consoleFull#12458680316baeffce-8504-4013-981e-7f26b5265816

**Comment by Shyam-Gupta** [comment]
Another hit today: https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_unit64_prtest/12223/consoleFull#-410211996a086b3e-df04-41d2-bc4d-43e8f9406d07

**Comment by Shyam-Gupta** [comment]
Same issue but with different error code this time: "xunit produced no error output but had exit code -1073740286"
https://ci.dot.net/job/dotnet_roslyn/job/dev15.7.x-vs-deps/job/windows_release_unit64_prtest/129/

https://ci.dot.net/job/dotnet_roslyn/job/dev15.7.x-vs-deps/job/windows_debug_unit64_prtest/136/

**Comment by 333fred** [comment]
Here as well: https://ci.dot.net/job/dotnet_roslyn/view/Official%20Builds/job/master/job/windows_debug_unit32/2095/
Ping @sharwell, have you looked into this at all?

**Comment by 333fred** [comment]
Signed build crashed with today: https://devdiv.visualstudio.com/DevDiv/_build?buildId=1475545&tab=details&_a=summary.

**Comment by sharwell** [comment]
@333fred build 2095 failed in TestExtensionErrorHandler after the following exception:

```
   at Microsoft.VisualStudio.Text.InterTextAdornmentSupport.Implementation.InterLineAdornmentManager.PerformLayout(SnapshotPoint trackingPoint)
   at Microsoft.VisualStudio.Text.InterTextAdornmentSupport.Implementation.InterLineAdornmentManager.OnBatchedTagsChanged(Object sender, BatchedTagsChangedEventArgs e)
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.RaiseEvent[TArgs](Object sender, EventHandler`1 eventHandlers, TArgs args)
```

**Comment by 333fred** [comment]
Failed again: https://ci.dot.net/job/dotnet_roslyn/view/Official%20Builds/job/dev15.7.x/job/windows_debug_unit32/176/console

**Comment by sharwell** [comment]
This will be fixed by #25560, which I can submit for review as soon as #25558 and #25620 are merged.
